optaplanner-core/src/main/java/org/optaplanner/core/api/domain/variable/StatefulVariableListener.java

```diff
@@ -14,15 +14,19 @@
  * limitations under the License.
  */
 
-package org.optaplanner.core.impl.domain.variable.listener;
+package org.optaplanner.core.api.domain.variable;
 
+import org.optaplanner.core.api.domain.variable.VariableListener;
 import org.optaplanner.core.api.score.director.ScoreDirector;
 import org.optaplanner.core.impl.domain.variable.descriptor.VariableDescriptor;
 import org.optaplanner.core.impl.domain.variable.supply.Supply;
 
 /**
  * A stateful {@link VariableListener},
  * often used to externalize data for a {@link Supply} from the domain model itself.
+ * <p>
+ * Implementation note: if/when any new method is added to this interface, make sure that
+ * {@link SupplyVariableListener} overrides it and delegates to it.
  */
 public interface StatefulVariableListener<Solution_, Entity_> extends VariableListener<Solution_, Entity_> {
```

Comment:
This will make `VariableDescriptor` leak to a public API.
(It leaked anyway, since people were using `StatefulVariableListener` anyway. This just makes it more "official".)

Reply:
Are many users using `StatefulVariableListener`? I don't think so. Let's not move `StatefulVariableListener` into the api package yet? When in doubt, leave it out?